### PR TITLE
Qualcomm AI Engine Direct - Reland GA Static QWEN2.5 0.5B

### PR DESCRIPTION
**Pull request by winskuo-quic**
### Summary
Previous PR were merged unintentionally: https://github.com/pytorch/executorch/pull/12054
On top of previous PR, also had a new commit addressing code review.

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

**Comment by winskuo-quic** [comment]
Hi @cccclai,
This PR is to reland the reverted PR: https://github.com/pytorch/executorch/pull/12054
I also have a new commit on top of original PR that is addressing issues discussed during code review, which includes:
1. Contributing to the CPU config instead of converting the config ourselves so that all backends can be using the config. I have created a new .json file for qwen2.5 0.5b since codebase now only has qwen2.5 1.5b config.
2. In previous PR, we mentioned that tokenizer went into runtime error on device. This issue is resolved; however, another issue related to `normalizer` shows up. I believe it is resolved in tokenizer's mainline: https://github.com/pytorch-labs/tokenizers/pull/104. However, since ExecuTorch is still using tokenizer without this patch(https://github.com/pytorch-labs/tokenizers/pull/104), it will still run into error. To work around, I have popped the `normalizer` key for now, and I can remove the workaround once ExecuTorch uses newer version of tokenizer that has the issue resolved.

File: examples/models/qwen2_5/config/0_5b_config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "hidden_dim": 4864,
+  "n_heads": 14,
+  "n_kv_heads": 2,
+  "n_layers": 24,
+  "norm_eps": 1e-06,
+  "rope_theta": 1000000.0,
+  "use_scaled_rope": false,
+  "vocab_size": 151936,
+  "use_hf_rope": true,
+  "attention_qkv_bias": true
```
Comment (cccclai) [inline examples/models/qwen2_5/config/0_5b_config.json:13]: Ah nice

File: examples/qualcomm/oss_scripts/llama/llama.py [hunk 2]
```diff
@@ -563,31 +578,47 @@ def compile(args, pte_filename, tokenizer):
...
-        state_dict[f"layers.{layer_i}.attention.wq.weight"] = permute(
-            state_dict[f"layers.{layer_i}.attention.wq.weight"], n_heads
+            checkpoint = download_and_convert_hf_checkpoint(model_id, convert_weights)
+        state_dict = torch.load(
+            checkpoint, weights_only=True, map_location="cpu", mmap=True
         )
-        state_dict[f"layers.{layer_i}.attention.wk.weight"] = permute(
-            state_dict[f"layers.{layer_i}.attention.wk.weight"], n_kv_heads
+    else:
+        state_dict = torch.load(
```
Comment (cccclai) [inline examples/qualcomm/oss_scripts/llama/llama.py:612]: Does it mean we support both huggingface model and static llama?

Comment (winskuo-quic) [reply examples/qualcomm/oss_scripts/llama/llama.py:612]: Thanks for reviewing the PR.
I think the use of huggingface's weight in static llama is there before this pr.
For Qwen, it's just going into the `if` statement. Static Llama's behavior is not changed.

**Comment by cccclai** [comment]
Hey can you rebase?

**Review by cccclai — APPROVED** [review]
Looks good, thank you!
